### PR TITLE
React servlet setup

### PR DESCRIPTION
**Pull request by Cindyzhang977**
Set up project with React and Java Servlet.

- Create button that displays text from servlet onClick
- Change front end port to 5000
- Set up react bootstrap
- Update README with setup instructions and dev resources

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+## General Setup
+```
+cd fontend
+npm install
+```
 
+## Development Resources
+* [React Bootstrap](https://react-bootstrap.github.io/getting-started/introduction/)
+* [Markdown Guide](https://guides.github.com/features/mastering-markdown/)
+* [Autoformat on Commit - Prettier](https://docs.google.com/document/d/1FQPR4w38ixA_ic0y0FEo_fQtRvLBiphEwx-XLEiLaTw/edit)
```
Comment (genedwards) [inline README.md:12]: I would remove this, I'll be adding the linter/formatter soon and this setup will be outdated and possibly in conflict with the new setup.

File: frontend/src/index.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
+import './styles/index.css';
```
Comment (genedwards) [inline frontend/src/index.js:3]: what's the goal of moving css to the styles directory? I would be against putting all css in a single directory.

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,29 +1,25 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (mecolmg) [inline frontend/src/App.js:2]: I'd prefer this to be in the index.js file like index.css

File: frontend/src/App.js [hunk 4]
```diff
@@ -1,29 +1,25 @@
...
 
 function App() {
-  fetch("/api/v1/data")
+  const [text, setText] = useState('');
+
+  function handleClick() {
+    fetch("/api/v1/data")
       .then((response) => response.text())
       .then((text) => {
         console.log("Here is the text from servlet: ", text);
```
Comment (mecolmg) [inline frontend/src/App.js:12]: Can you remove the console.log now that you're doing something with the text. You can then make this whole thing:

```js
fetch("/api/v1/data")
  .then((response) => response.text())
  .then(setText);
```

**Pull request merged**